### DISCUSSION TITLE
Inconstant results

### DISCUSSION THREAD

**Discussion by T-prog3** [TumblThreeApp/TumblThree#280] [General]
1. So i downloaded the same twitter blog multiple times within a 15 minute gap and noticed that i got two different results. 
2. The first time i got **1910 .jpg files** and **286 .mp4 files**. 
3. I then removed the blog from TumblThree and renamed the blog folder. 
4. Then i added the same blog again
5. This second time i got the same number of images **1910**, but now i got **69** more video files in a total of **355**
6. So i tried a third time
7. Now i did get the exact same result 1910 and 355.

I've tried to replicate the first result by tweaking settings but nothing will bring me to the same result, The closest is to enable the `skip .gif files`. This took me almost to the first result and gave me 280 video files, but still off by 6. This would be a possible reason if it weren't the case that all but one of the **69** video files that differ have audio. So they are not .gif files.

I still don't understand why there could be such a big difference in such a short time when no settings have been changed. But there seem to be an issue somewhere in relation to the video file downloads.

**Comment by thomas694** [comment]
Oh, that's an easy one. It has to do with the way these platforms store and retrieve their data. You can experience it on many similar platforms, also on Tumblr.
No human is scrolling through a blog so fast, viewing every post/media and/or goes back so much in time when viewing the blog in the browser, or would be able to see if they effectively scrolled through e.g. 500 or 520 posts.

TumblThree downloads very fast and in parallel, back in time until the blog's first post. So when requested [by browser/TumblThree], older/bigger media files aren't all available instantly, they are put in a queue for retrieval from their storage systems and will be available sometime later.
Therefore the option "Force rescan" is helpful. Just run a newly added blog after a couple of minutes/hours (sometimes a bit longer) again and it will download more items on some blogs. That depends on factors like how old/big the post/media is, how many users wanted to view it recently and so on.

Of course, that was simplified, but it describes the basic things.

**Comment by T-prog3** [comment]
Thanks for the information, i didn't know that! Will do a lot of force rescans henceforth.
